### PR TITLE
[IMP] web: add include archived to domain selector

### PR DESCRIPTION
**Pull request by juliusc2066**
This commit introduces a new feature to the domain selector that allows to toggle a new "include archived" option that alters the domain in a way such that archived records are included or not. This domain part is naive and may overlap with custom parts of the domain. The special domain part which includes archived records (['&', <previous_domain>, '|', ('active' , '=', True), ('active', '=', False)]) is invisible inside the domain tree but remains visible in the domain code editor.

task-3482381

### CODE REVIEW

File: addons/web/static/src/core/domain_selector/domain_selector.js [hunk 1]
```diff
@@ -177,16 +193,47 @@ export class DomainSelector extends Component {
             options
         ).children[0];
 
+        this.includeArchived = false;
+        if (this.tree.value === "&") {
```
Comment (Polymorphe57) [inline addons/web/static/src/core/domain_selector/domain_selector.js:197]: is this necessary if showArchivedCheckbox is false?

Comment (juliusc2066) [reply addons/web/static/src/core/domain_selector/domain_selector.js]: It is not indeed

File: addons/web/static/src/core/domain_selector/domain_selector.js [hunk 2]
```diff
@@ -177,16 +193,47 @@ export class DomainSelector extends Component {
             options
         ).children[0];
 
+        this.includeArchived = false;
+        if (this.showArchivedCheckbox && this.tree.value === "&") {
```
Comment (Polymorphe57) [inline addons/web/static/src/core/domain_selector/domain_selector.js:197]: not only here...

File: addons/web/static/src/core/domain_selector/domain_selector.xml [hunk 3]
```diff
@@ -4,23 +4,44 @@
...
-                    <t t-if="!props.readonly">
-                        <button class="btn btn-sm btn-primary o_domain_add_first_node_button ms-1" t-on-click="() => this.insertRootLeaf(node)">
-                            <i class="fa fa-plus"/> Add condition
-                        </button>
-                    </t>
+                    <div class="d-flex justify-content-between">
+                        <div>
+                            <span>Match <strong>all records</strong></span>
+                        </div>
+                        <CheckBox
```
Comment (Polymorphe57) [inline addons/web/static/src/core/domain_selector/domain_selector.xml:13]: Can't you factorize a bit the template with a t-call?

**Comment by Polymorphe57** [comment]
robodoo r+

**Pull request closed without merging**